juju/juju
# Caas scale applications

## Description of change

For k8s models, add/remove unit is replaced with a scale-application command.
The work is broken up into several commits:

1. State - Add application scale attribute and a watcher for scale changes
2. APIServer - Add application scale methods to facades and make caas unit provisioner react to added pods again.
3. API - Add application scale methods to facade clients
4. Modify caas provisioner worker to react to scale changes
5. k8s client - Delete all pods when scale becomes 0
6 CLI - Add application-scale command and make add-unit/remove-unit IAAS only

## QA steps

bootstrap a k8s model
deploy an application
scale up and down and check juju status and juju storage
